doc/openal-support.dox

```diff
@@ -0,0 +1,93 @@
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @page openal-support OpenAL support state
+@brief List of (un)supported OpenAL features and extensions.
+
+@tableofcontents
+
+@section openal-support-state OpenAL implementation state
+
+The extension implementation is considered complete if all its defined types,
+functions and enum values are exposed through the API.
+
+
+@subsection openal-extension-support Extensions
+
+Extension                                   | Status
+------------------------------------------- | ------
+@alc_extension{ENUMERATION,ALL_EXT}         | |
+@alc_extension{ENUMERATION,EXT}             | |
+@alc_extension{EXT,CAPTURE}                 | |
+@alc_extension{LOKI,audio_channel}          | |
+@alc_extension{LOKI,quadriphonic}           | |
+@alc_extension{EXT,STATIC_BUFFER}           | |
+@alc_extension{EXT,ASA}                     | |
+@alc_extension{EXT,ASA_ROGER_BEEP}          | |
+@alc_extension{EXT,ASA_DISTORTION}          | |
+@alc_extension{EXT,EFX}                     | |
+@al_extension{EXT,MP3}                      | |
+@al_extension{EXT,vorbis}                   | |
+@al_extension{EXT,DEDICATED}                | |
+@al_extension{EXT,DEFAULT_FILTER_ORDER}     | |
+@al_extension{EXT,disconnect}               | |
+@al_extension{EXT,double}                   | done
+@al_extension{EXT,float32}                  | done
+@al_extension{EXT,ALAW}                     | |
+@al_extension{EXT,IMA4}                     | |
+@al_extension{EXT,MULAW}                    | |
+@al_extension{EXT,MULAW_BFORMAT}            | |
+@al_extension{EXT,MULAW_RADIUS}             | |
+@al_extension{EXT,MCFORMATS}                | |
+@al_extension{EXT,FOLDBACK}                 | |
+@al_extension{EXT,SOURCE_RADIUS}            | |
+@al_extension{EXT,STEREO_ANGLES}            | |
+@al_eax_extension{2.0}                      | |
+@al_eax_extension{3.0}                      | |
+@al_eax_extension{4.0}                      | |
+@al_eax_extension{5.0}                      | |
+@al_eax_extension{_RAM}                     | |
+
+@subsection openal-extension-support OpenAL Soft Extensions
+
+Extension                                   | Status
+------------------------------------------- | ------
+@alc_extension{SOFTX,HRTF}                  | done
+@alc_extension{SOFTX,device_clock}          | |
+@alc_extension{SOFTX,midi_interface}        | |
+@alc_extension{SOFT,loopback}               | |
+@alc_extension{SOFT,pause_device}           | |
+@al_extension{SOFT,block_alignment}         | |
+@al_extension{SOFT,buffer_samples}          | |
+@al_extension{SOFT,buffer_sub_data}         | |
+@al_extension{SOFT,deferred_updates}        | |
+@al_extension{SOFT,direct_channels}         | |
+@al_extension{SOFT,loop_points}             | |
+@al_extension{SOFT,MSADPCM}                 | |
+@al_extension{SOFT,source_latency}          | |
+@al_extension{SOFT,source_length}           | |
```

Comment:
I would suggest listing only those that are supported now, otherwise this might look like we commited ourselves to support every one of them, which I think is not the case.

With OpenGL I'm listing all extensions that I'm willing to/able to implement -- i.e. all extensions that are part of any core version, all core ARB/KHR extensions and some (but not all) selected vendor extensions on top of these.